### ISSUE TITLE
Removing and readding a 3D image layer is broken

### ISSUE THREAD

**Issue by brisvag** [napari/napari#3667]
## 🐛 Bug

Re-adding a previously removed 3D image layer results in the following error and a failure to add the layer:

```python
ValueError: interpolation must be one of bessel, bicubic, bilinear, blackman, catrom, gaussian, hamming, hanning, hermite, kaiser, lanczos, mitchell, nearest, quadric, sinc, spline16, spline36
```

This only happens with *3D images*. 2D images are fine.

As appropriate, `git bisect` tells me that it was a regression introduced by *me*, over at #3333 :P

## To Reproduce

```python
import napari
import numpy as np

v = napari.Viewer(ndisplay=3)
img = np.random.rand(100, 100, 100)
il = napari.layers.Image(img)
v.add_layer(il)
# everything is fine so far
```
```python
v.layers.remove(il)
v.add_layer(il)
# error!
```

**Comment by kevinyamauchi** [comment]
> As appropriate, git bisect tells me that it was a regression introduced by me, over at #3333 :P

Haha, git bisect can be so cruel 😛

**Comment by kevinyamauchi** [comment]
Is this related? https://github.com/napari/napari/issues/3551

**Comment by alisterburt** [comment]
just checked and #3551 is still present with the proposed fix at #3670 so probably related but separate

**Comment by brisvag** [comment]
#3551 is not related. I explained why there!
